src/test/java/core/basesyntax/service/WriterServiceImplTest.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.impl.WriterServiceImpl;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class WriterServiceImplTest {
+    private static final String TEST_RESOURCES_PATH = "src/test/resources/";
+    private static WriterService writerService;
+
+    @BeforeClass
+    public static void setUp() {
+        writerService = new WriterServiceImpl();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void write_invalidFilePath_notOk() {
+        String report = "Test report content";
+        String invalidFilePath = TEST_RESOURCES_PATH + "\0-invalid-.txt";
+        writerService.write(report, invalidFilePath);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void write_nullReport_notOk() {
+        String report = "test report";
+        String fileName = TEST_RESOURCES_PATH + "test-report.txt";
+        writerService.write(null, fileName);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void write_nullFilePath_notOk() {
```

Comment:
let's check ok case
after writing, read content of the file with Files.readString​() and compare with your report